From eb555bcc219bdd332d0bff45aa66c26f7a846307 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20M=C3=BCller?= <michael@fds-team.de>
Date: Sun, 21 Jun 2015 20:04:34 +0200
Subject: configure.ac: Remove check for strength-reduce bug.

This was bug was fixed in gcc-2.7.2.1 which was released 19 years ago.
When cross compiling, the configure check currently assumes that the bug
is present, which produces a lot of noise with non-gcc compilers.
---
 configure.ac | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3587a12..fa6a542 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1788,22 +1788,6 @@ if test "x${GCC}" = "xyes"
 then
   EXTRACFLAGS="-Wall -pipe"
 
-  dnl Check for strength-reduce bug
-  AC_CACHE_CHECK( [for gcc strength-reduce bug], ac_cv_c_gcc_strength_bug,
-                  AC_RUN_IFELSE([AC_LANG_PROGRAM([[int L[[4]] = {0,1,2,3};]],
-[[static int Array[[3]];
-  unsigned int B = 3;
-  int i;
-  for(i=0; i<B; i++) Array[[i]] = i - 3;
-  for(i=0; i<4 - 1; i++) L[[i]] = L[[i + 1]];
-  L[[i]] = 4;
-  return (Array[[1]] != -2 || L[[2]] != 3)]])],
-    [ac_cv_c_gcc_strength_bug="no"],[ac_cv_c_gcc_strength_bug="yes"],[ac_cv_c_gcc_strength_bug="yes"]) )
-  if test "$ac_cv_c_gcc_strength_bug" = "yes"
-  then
-    EXTRACFLAGS="$EXTRACFLAGS -fno-strength-reduce"
-  fi
-
   dnl Check for some compiler flags
   WINE_TRY_CFLAGS([-fno-builtin],[AC_SUBST(BUILTINFLAG,"-fno-builtin")])
   WINE_TRY_CFLAGS([-fno-strict-aliasing])
-- 
2.4.3

